doorstop/run_bepinex.sh

```diff
@@ -5,73 +5,87 @@
 
-doorstop_libs=${PWD}/doorstop_libs;
-arch="";
-executable_path="";
-lib_postfix="";
+doorstop_libs="${PWD}/doorstop_libs"
+arch=""
+executable_path=""
+lib_postfix=""
 
-os_type=`uname -s`;
+os_type=`uname -s`
 case $os_type in
-    Linux*)     executable_path=${PWD}/${executable_name};
-                lib_postfix="so";;
-    Darwin*)    executable_path=${PWD}/${executable_name}.app/Contents/MacOS/${executable_name};
-                lib_postfix="dylib";;
-    *)          echo "Cannot identify OS (got $(uname -s))!";
-                echo "Please create an issue at https://github.com/BepInEx/BepInEx/issues."; 
-                exit 1;;
+    Linux*)
+        executable_path="${PWD}/${executable_name}"
+        lib_postfix="so"
+        ;;
+    Darwin*)
+        executable_path="${PWD}/${executable_name}.app/Contents/MacOS/${executable_name}"
+        lib_postfix="dylib"
+        ;;
+    *)
+        echo "Cannot identify OS (got $(uname -s))!"
+        echo "Please create an issue at https://github.com/BepInEx/BepInEx/issues."
+        exit 1
+        ;;
 esac
 
 # Special case: if there is an arg, use that as executable path
 # Linux: arg is path to the executable
 # MacOS: arg is path to the .app folder which we need to resolve to the exectuable
 if [ -n "$1" ]; then
     case $os_type in
-        Linux*)     executable_path=$1;;
-        Darwin*)    executable_name=`basename "$1" .app`;
-                    executable_path=$1/Contents/MacOS/$executable_name;;
+        Linux*)
+            executable_path="$1"
+            ;;
+        Darwin*)
+            executable_name=`basename "$1" .app`
+            executable_path="$1/Contents/MacOS/$executable_name"
+            ;;
     esac
 fi
 
-executable_type=`file -b "${executable_path}"`;
+executable_type=`LD_PRELOAD="" file -b "${executable_path}"`;
+
 case $executable_type in
-    *64-bit*)           arch="x64";;
-    *32-bit*|*i386*)    arch="x86";;
-    *)          echo "Cannot identify executable type (got ${executable_type})!"; 
-                echo "Please create an issue at https://github.com/BepInEx/BepInEx/issues."; 
-                exit 1;;
+    *64-bit*)
+        arch="x64"
+        ;;
+    *32-bit*|*i386*)
+        arch="x86"
+        ;;
+    *)
+        echo "Cannot identify executable type (got ${executable_type})!"
+        echo "Please create an issue at https://github.com/BepInEx/BepInEx/issues."
+        exit 1
+        ;;
 esac
 
-doorstop_libname=libdoorstop_${arch}.${lib_postfix};
-export LD_LIBRARY_PATH=${doorstop_libs}:${LD_LIBRARY_PATH};
-export LD_PRELOAD=${doorstop_libs}/$doorstop_libname:$LD_PRELOAD_BAK;
-export DYLD_LIBRARY_PATH=${doorstop_libs};
-export DYLD_INSERT_LIBRARIES=${doorstop_libs}/$doorstop_libname;
+doorstop_libname=libdoorstop_${arch}.${lib_postfix}
+export LD_LIBRARY_PATH="${doorstop_libs}":${LD_LIBRARY_PATH}
+export LD_PRELOAD=$doorstop_libname:$LD_PRELOAD
+export DYLD_LIBRARY_PATH="${doorstop_libs}"
+export DYLD_INSERT_LIBRARIES=$doorstop_libname
```

Comment:
On macOS it seems that you need to specify the full path to the library into `DYLD_INSERT_LIBRARIES` as it does not properly respect `DYLD_LIBRARY_PATH`. Unlike linux, macOS handles spaces fine, so you can revert it to
```sh
export DYLD_LIBRARY_PATH=${doorstop_libs}
export DYLD_INSERT_LIBRARIES=${doorstop_libs}/$doorstop_libname
```